platform/features/conductor-redux/src/README.md

```diff
@@ -0,0 +1,187 @@
+//Somewhere in the TimeConductorController...
+function changeTOI(newTime) {
+    MCT.conductor.timeOfInterest(newTime);
+}
+```
+### 3. Plot controller listens for change to TOI
+```javascript
+// toi is attribute of Time Conductor object. Add a listener to the time 
+// conductor to be alerted to changes in value
+
+MCT.conductor.listen(function (conductor) {
+    var toi = conductor.timeOfInterest();
+    plot.setTimeOfInterest(toi);
+});
+```
+### 4. Real-time object is loaded and time conductor is set up in real-time mode showing last 15 minutes of data.
+```javascript
+// Notional initialization function, perhaps in a 'runs' extension
+function initializeConductor(telemetryMetadata) {
+    var tc = MCT.conductor,
+        FIFTEEN_MINUTES = 15 * 60 * 1000;
+    tc.timeSystem(telemetryMetadata.timeSystem);
+    
+    //Start and end modes set separately (to support fixed start, follow 
+    // time end. Follow time = RELATIVE end mode where delta value = 0) 
+    tc.startMode(tc.Modes.RELATIVE);
+    tc.endMode(tc.Modes.RELATIVE);
+    
+    // Delta defined in ms. Start time delta is relative to end time.
+    tc.bounds.inner.start.delta(FIFTEEN_MINUTES);
+    tc.bounds.outer.start.delta(FIFTEEN_MINUTES);
+    
+    // Delta defined in ms. An end time delta value is relative to 'now'. 
+    // Thus, an end delta of zero is 'follow time'. May be > 0 to track 
+    // time in the future.
+    // Could add a convenience function to TC to make setting follow 
+    // time mode a little clearer
+    tc.bounds.inner.end.delta(0);
+    tc.bounds.outer.end.delta(0);
+}
+```
+### 5. Switch to follow time. Set inner end to now, and outer end to now + 1 hour
+
+```javascript
+
+var tc = MCT.conductor;
+
+// Somewhere in the time conductor controller...
+function setFollowTime() {
+    var ONE_HOUR = 1 * 60 * 60 * 1000;
+    
+    tc.endMode(tc.bounds.Modes.RELATIVE);
+    tc.startMode(tc.bounds.Modes.RELATIVE);
+    
+    //Set start deltas. Start deltas always relative to end
+    tc.bounds.inner.start.delta(-ONE_HOUR);
+    tc.bounds.outer.start.delta(-ONE_HOUR);
+    
+    //Set end deltas. End deltas always relative to `now`
+    tc.bounds.inner.end.delta(0);
+    tc.bounds.outer.end.delta(ONE_HOUR);
+}
+```
+
+### 6. Plot Controller updated on tick
+``` javascript
+// Plot only really interested in inner bounds.
+// Specify event type of TICK to receive tick events only.
+MCT.conductor.bounds.inner.listen(function (innerBounds) {
+    plotUpdater.setDomainBounds(bounds.start.time(), bounds.end.time());
+}, MCT.conductor.bounds.EventTypes.TICK);
+```
+
+### 7. Plot Controller updated when user changes bounds
+``` javascript
+// Plot only really interested in inner bounds
+// Specify event type of USER to receive user events (such as bounds 
+// handles being dragged) only.
+MCT.conductor.bounds.inner.listen(function(bounds) {
+    plotUpdater.setDomainBounds(bounds.start.time(), bounds.end.time());
+}, MCT.conductor.bounds.EventTypes.USER);
```

Comment:
Not to stray too far out of scope, but this looks like a suitable opportunity to make a broader design decision around change notification. This appears to extend our current approach, which I am a little hesitant to do when there are third-party options available. Additionally, looking at the API redesign there are a lot of cases where we'll want to provide some form of change notification for objects (especially true as we rely less on Angular and the digest cycle to detect changes for us)

There are three paths I see as viable here:

1. [EventEmitter](https://nodejs.org/api/events.html) (or, more specifically, one of the browser implementations)
  * Pros:
    * Very standard; since this API is built into NodeJS it will seem native to some developers.
    * Multiple implementations are available to pick and choose from.
    * JSDoc has a tag for [events](http://usejsdoc.org/tags-event.html), making this fairly documentable
    * Allows the same object to fire multiple event types (as is done with this `listen` call)
  * Cons:
    * Based on string-matching, which we've generally wanted to avoid. (But, one of the detriments of string-matching based approaches is the difficulty in documenting them, which JSDoc support mitigates)
    * More methods are exposed than we might generally wish to (and some implementations have even more.) In particular, `emit` is exposed everywhere.
2. Use observables/reactive programming (e.g. [Kefir](https://rpominov.github.io/kefir), [RxJS](https://github.com/Reactive-Extensions/RxJS), [Bacon](https://baconjs.github.io/))
  * Pro: Include support for chaining observables in various ways (maps, filters, etc), which could greatly simplify the backend-to-view data flow.
  * Con: Non-standard, and tend to expose rather large APIs. Hard to change this decision later, esp. if we expose these as part of our public API (and listening for changes is a public API use case)
3. Extend our existing event system
  * Pros: We get exactly what we want and nothing more.
  * Cons: More effort, testing, maintenance required.

Of the options, EventEmitter certainly looks like the safest choice. I think the other options are "better" in certain respects (reactive programming is powerful, and having a small API that does exactly what we need is great, too) but it would be hard to argue that they're better enough to justify the investment/risk